lib/ykpiv.c

```diff
@@ -704,6 +761,10 @@ ykpiv_rc ykpiv_verify(ykpiv_state *state, const char *pin, int *tries) {
   if((res = send_data(state, &apdu, data, &recv_len, &sw)) != YKPIV_OK) {
     return res;
   } else if(sw == SW_SUCCESS) {
+    if (state->pin == NULL) {
```

Comment:
This check is a little bit problematic in that it wont update the state->pin if ykpiv_verify is called again with a different pin, it still saves the old one in state->pin (we might also want to consider if state->pin should be updated on pin change).